### PR TITLE
Adding OpenStack deployment as part of the JSON output

### PR DESCRIPTION
**Pull request by jsanemet**
- Extracted 'SerializedPrinter' and 'JSONPrinter' to be their own independent thing. On both, added the 'Config' class so that it is possible to reuse the arguments on other printers.
- Extracted serializing functions into 'SerializedPrinter.SerializationProvider'. Implementing this class is enough to have new output formats. See 'JSONPrinter' for an example.
- Reorganized inheritance of classes below 'SerializedPrinter' and 'JSONPrinter'. Prefixed them with 'CI' to avoid naming conflicts.
- Reworked how retrieval of plugin section is made. Cibyl no longer needs to know about the plugin's inners, instead, it expects the plugin to provide a 'PluginPrinterTemplate'. Cibyl will just call the correct output style and go on with what the plugin returns. The printer instance is installed in the model much like the functions are.
- Implemented 'OSJSONPrinter' to print the deployment as JSON.

### CODE REVIEW

File: cibyl/plugins/openstack/printers/serialized.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                'tls_everywhere': network.tls_everywhere.value
+            },
+            'storage': {
+                'cinder_backend': storage.cinder_backend.value
+            },
+            'ironic': {
+                'ironic_inspector': ironic.ironic_inspector.value,
+                'cleaning_network': ironic.cleaning_network.value
+            },
+            'overcloud_templates': list(deployment.overcloud_templates.value),
```
Comment (cescgina) [inline cibyl/plugins/openstack/printers/serialized.py:59]: `overcloud_templates` could be `None` and this line would crash. I got the error running `cibyl -vv -f json --config jenkins.yaml query --jobs DFG-network-openstack-designate-17.0_director-rhel-virthost-3cont_2comp-ipv4-geneve --last-build --ip-version --cinder-backend` against downstream Jenkins

File: cibyl/plugins/openstack/printers/serialized.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                for stage in deployment.stages
+            ]
+        }
+
+        return self.provider.dump(result)
+
+    @overrides
+    def print_test_collection(self, collection) -> str:
+        result = {
+            'tests': list(collection.tests.value),
```
Comment (cescgina) [inline cibyl/plugins/openstack/printers/serialized.py:88]: This could have the same issue as the overcloud_templates line

Comment (jsanemet) [reply cibyl/plugins/openstack/printers/serialized.py]: Fixed.

**Review by cescgina — APPROVED** [review]
Looks great, very clean solution to use the `PluginPrinterTemplate`to decouple the core printer from the plugin ones. +1

**Review by adrianfusco — APPROVED** [review]
+1

**Pull request merged**